inferno-core/src/Inferno/Infer/Exhaustiveness.hs

```diff
@@ -246,3 +249,20 @@ instance Enum EnumText where
       if Text.null t
         then "a"
         else t <> t
+
+-- | A type for array patterns. Since this is used only for its Enum instance, in order to
+-- generate counter-examples for incomplete pattern matches, we map any array pattern to the
+-- number of items in it.  For any pattern [<bla>], succ generates the pattern [<bla>, _].
+newtype EnumArrayPat = EnumArrayPat Int
+  deriving (Show)
+
+mkEnumArrayPat :: [(Pat a SourcePos, Maybe SourcePos)] -> EnumArrayPat
+mkEnumArrayPat = EnumArrayPat . length
+
+instance Pretty EnumArrayPat where
+  pretty (EnumArrayPat n) =
+    pretty $ PArray undefined (take n $ repeat $ (PVar (initialPos "") Nothing, Nothing)) undefined
```

Comment:
Are the `undefined`s intentional? If so, maybe a comment would be helpful

Reply:
Also, `take n . repeat = replicate n`

Reply:
I like `error "something searchable"` vs `undefined` for ease of debugging in case the impossible happens